src/unitquaternion.jl

```diff
@@ -365,6 +365,43 @@ function kinematics(q::Q, ω::AbstractVector) where Q <: UnitQuaternion
     1//2 * params(q*Q(0.0, ω[1], ω[2], ω[3], false))
 end
 
+# ~~~~~~~~~~~~~~~ Spherical Linear Interpolation ~~~~~~~~~~~~~ #
+
+function slerp(qa::UnitQuaternion{T}, qb::UnitQuaternion{T}, t, ϵ=0.001) where T
+    # http://www.euclideanspace.com/maths/algebra/realNormedAlgebra/quaternions/slerp/
+    coshalftheta = qa.w * qb.w + qa.x * qb.x + qa.y * qb.y + qa.z * qb.z;
+
+    if coshalftheta < 0
+        qm = -qb
+        coshalftheta = -coshalftheta
+    else
+        qm = qb
+    end
+    abs(coshalftheta) >= 1.0 && return qa
+
+    halftheta    = acos(coshalftheta)
+    sinhalftheta = sqrt(one(T) - coshalftheta * coshalftheta)
+
+    if abs(sinhalftheta) < ϵ
+        return UnitQuaternion(
+            T(0.5) * (qa.w + qb.w),
+            T(0.5) * (qa.x + qb.x),
+            T(0.5) * (qa.y + qb.y),
+            T(0.5) * (qa.z + qb.z),
```

Comment:
```suggestion
            (qa.w + qb.w)/2,
            (qa.x + qb.x)/2,
            (qa.y + qb.y)/2,
            (qa.z + qb.z)/2,
```
This is more readable, and there is no performance regression, I guess.

Reply:
Yes it seems that the compiler knows that when a literal power of two is being divided by this is exactly equivalent to multiplying by its inverse (in IEEE floating point):

```
julia> foo(x,y) = (x+y)/2
foo (generic function with 1 method)

julia> @code_native foo(2.0,3.0)
	.text
; ┌ @ REPL[7]:1 within `foo`
; │┌ @ float.jl:399 within `+`
	vaddsd	%xmm1, %xmm0, %xmm0
	movabsq	$.rodata.cst8, %rax
; │└
; │┌ @ promotion.jl:382 within `/` @ float.jl:408
	vmulsd	(%rax), %xmm0, %xmm0
; │└
	retq
	nopw	%cs:(%rax,%rax)
; └
```

This isn't true for `foo(x,y) = (x+y)/3` though. Or I guess any other non-powers of two whose inverses aren't exactly representable (see https://stackoverflow.com/questions/64401669/are-there-ieee-754-floating-point-values-x-whose-inverted-values-1-x-are-also-ex)